encrypter.cpp

```diff
@@ -0,0 +1,190 @@
+#include <iostream>
+#include <string.h>
+#include <fstream>
+#include <ios>
+#include <random>
+#include <sstream>
+
+using Byte = char;
+
+//=====put_error=====//
+void put_error(const std::string& error_message)
+{
+  std::cout << error_message;
+  exit(1);
+}
+
+//=====shuffle=====//
+void shuffle(std::vector<Byte>& input)
+{
+  std::default_random_engine generator;
+  std::uniform_int_distribution<int> distribution(0, input.size() - 1);
+  int i = 0;
+
+  for (std::vector<Byte>::iterator iter = input.begin(); iter < input.end(); ++i, ++iter)
+  {
+    int random = distribution(generator);
+    Byte tmp = input[i];
+    input[i] = input[random];
+    input[random] = tmp;
+  } 
+}
+
+//=====replace_extension=====//
+std::string replace_extension(const std::string& filename, const std::string& new_extension)
+{
+  int i;
+
+  for (i = 0; filename[i] != '.'; ++i);
```

Comment:
it can be replaced by one of string::find..... functions
and it looks like it a bug in case if filename has more than one '.' which is quite usual in linux